### PR TITLE
JDBC: Created the retrieveGameStats() method that queries the databas…

### PR DESCRIPTION
**Pull request by navaell**
Created a method to query the database and retrive the overall game stats and returns them in a RetrievedGameStatistics object. Also wrote the class for that object.

### CODE REVIEW

File: src/main/java/model/Database.java [hunk 1]
```diff
@@ -70,18 +84,68 @@ public static void uploadGameStats(int draws, int rounds, String winner, Player[
...
+    }
+
+    /**
+     * Method creates an object that contains the retrieved overall game stats (5 metrics)
+     * @return object containing stats
+     */
+    public static RetrievedGameStatistics retrieveGameStats() {
+        RetrievedGameStatistics retrievedGameStatistics = null;
+        try {
+            int gamesPlayed, aiWins, userWins, maxRounds;
```
Comment (gsears) [inline src/main/java/model/Database.java]: Codacy Issue found: [Use one line for each declaration, it enhances code readability.](https://app.codacy.com/gh/shkembe-chorba/MScIT_TeamProject/pullRequest?prid=4870991)

**Comment by gsears** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- src/main/java/model/RetrievedGameStatistics.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/shkembe-chorba/MScIT_TeamProject/pullRequest?prid=4870991&bid=16148949)

File: src/main/java/model/Database.java [hunk 2]
```diff
@@ -14,7 +14,22 @@
...
+                new Player[]{
+                        new Player("USER"),
+                        new Player("AI1"),
+                        new Player("AI2"),
+                        new Player("AI3"),
+                        new Player("AI4")
+                }
+        );*/
+        // tests for retrieving stats from database
+        final RetrievedGameStatistics statistics = retrieveGameStats();
```
Comment (gsears) [inline src/main/java/model/Database.java]: Codacy Issue found: [Avoid unused local variables such as 'statistics'.](https://app.codacy.com/gh/shkembe-chorba/MScIT_TeamProject/pullRequest?prid=4870991)

File: src/main/java/model/Database.java [hunk 3]
```diff
@@ -70,18 +84,68 @@ public static void uploadGameStats(int draws, int rounds, String winner, Player[
...
+        RetrievedGameStatistics retrievedGameStatistics = null;
+        try {
+            int gamesPlayed, aiWins, userWins, maxRounds;
+            double avgDraws;
+
+            Statement statement = connection.createStatement();
+
+            //Obtains the number of games played overall
+            String sqlStringMaxGid = "SELECT max(gid) FROM game";
+            ResultSet result = statement.executeQuery(sqlStringMaxGid);
```
Comment (gsears) [inline src/main/java/model/Database.java]: Codacy Issue found: [Always check the return of one of the navigation method (next,previous,first,last) of a ResultSet.](https://app.codacy.com/gh/shkembe-chorba/MScIT_TeamProject/pullRequest?prid=4870991)

**Comment by gsears** [comment]
Heyy 😄 

Is it safe to delete this branch if it's stale (and merged into database)?

**Comment by navaell** [comment]
Sure thing. :)

**Pull request merged**